client/db/src/lib.rs

```diff
@@ -127,6 +127,24 @@ enum DbExtrinsic<B: BlockT> {
 	Full(B::Extrinsic),
 }
 
+/// Provide an access to the StateDb's metadata
+pub trait StateDbMeta {
+	/// Effectively used PruningMode
+	fn state_pruning(&self) -> PruningMode;
+}
```

Comment:
Why did you not moved it to the `client-api` crate as I proposed? Was there any problem with doing this?

Reply:
I misunderstood you then: I assumed you do not like it be located in a separate module.

I'll think of moving it to somewhere, so it is scattered among less places.

Reply:
The other traits are also defined there. So, I wanted to have them in a common place.

Reply:
Added a method: `fn requires_full_sync(&self) -> bool` to the `sc-client-api::BlockBackend` and `sc-client-api::Backend`, so that there would be no need to depend on `PruningMode` in `sc-client-api` crate.